### ISSUE TITLE
Can't choose condition suggestions with keyboard

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#2163]
This is a regression caused by #2158: now if you focus a condition in the autocomplete suggestion list and press <kbd>Tab</kbd> or <kbd>Enter</kbd>, the autocomplete will complete whatever you typed in the box, rather than completing the focused condition:

![image](https://user-images.githubusercontent.com/37659591/70712984-fced1480-1cdc-11ea-909a-63c82038b29d.png)

You can still pick suggested conditions using the mouse, but I'm marking this as high priority in case anybody else uses the keyboard for completing experimental conditions. If this proves to be a big problem, then we should revert #2158 for now.

**Comment by jseager7** [comment]
I've pushed a quick fix for this. It's a pretty awful hack involving jQuery selectors, but I couldn't figure out anything better. Essentially we scan the suggestions list to see if the user has focused any terms before adding a term, and if there is a focused term, that gets added instead of what the user typed:

```js
if (!(that.options.autocomplete.autoFocus && that.tagInput.data('autocomplete-open'))) {
    var focusedItem = that.tagInput.autocomplete('widget').find('.ui-menu-item > .ui-state-focus');
    if (focusedItem.length > 0) {
        that.createTag(focusedItem[0].innerText);
    } else {
        that.createTag(that._cleanedInput());
    }
    that.tagInput.autocomplete('close');
}
```

Based on manual testing, it doesn't seem to break adding custom terms, nor does it break selecting terms using the mouse, so I'll close this issue as fixed.

@kimrutherford Next time you're around, it might be worth seeing if we can come up with a cleaner fix. There's probably a better way; it just depends if the effort is worth it.

**Comment by kimrutherford** [comment]
> It's a pretty awful hack

I'm a fan of awful hacks that fix bugs.

> Next time you're around, it might be worth seeing if we can come up with a cleaner fix.

Sounds good.  Is this hack needed in the latest tagit.js?

**Comment by jseager7** [comment]
> Is this hack needed in the latest tagit.js?

Yes, because it's a regression bug introduced by the fix in #2158, which is merged in the upstream tag-it.js repository (so the bug presumably exists upstream too). I couldn't see any fixes suggested for this on their tracker, so I based this fix on the one in #2162.
